packages/websocket/src/abstract-websocket.client.ts

```diff
@@ -96,26 +87,42 @@ export abstract class AbstractWebsocketClient<K> implements WebsocketHandler<K>,
         this.disconnect();
     }
 
+    protected serialize(message: Any): WebSocketMessage {
+        return JSON.stringify(message);
+    }
+
+    protected deserialize(messageEvent: MessageEvent): WebsocketMessage<K, Any> {
+        return JSON.parse(messageEvent.data);
+    }
+
+    private onOpenObserver(event: Event): void {
+        this.connected = true;
+        this.connected$.next(event);
+        this.onConnected(event);
+    }
+
+    private onCloseObserver(event: Event): void {
+        this.connected = false;
+        this.disconnected$.next(event);
+        this.onDisconnected(event);
+    }
+
     private _connect(): void {
         this.socket$ = webSocket(this.webSocketSubjectConfig);
-        this.subscription = this.socket$
-            .pipe(
-                catchError(
-                    (err: Error): Observable<never> => {
-                        this.reconnection(this.subscription);
-                        return throwError(err);
-                    }
-                ),
-                takeUntil(this.destroy$)
-            )
-            .subscribe((message: WebsocketMessage<K, Any>): void => {
+        this.subscription = this.socket$.pipe(takeUntil(this.destroy$)).subscribe(
+            (message: WebsocketMessage<K, Any>): void => {
                 window.requestAnimationFrame((): void => this.messages$.next(message));
```

Comment:
What do you think about remove `requestAnimationFrame`?) 

can we somehow correctly throttling the event? so as not to interfere with the events of change detection